x/gamm/keeper/swap_test.go

```diff
@@ -270,3 +272,47 @@ func (suite *KeeperTestSuite) TestActiveBalancerPoolSwap() {
 		}
 	}
 }
+
+// Test two pools -- one is active and should have swaps allowed,
+// while the other is inactive and should have swaps frozen.
+func (suite *KeeperTestSuite) TestInactivePoolFreezeSwaps() {
+	// Setup test
+	suite.SetupTest()
+	testCoin := sdk.NewCoin("foo", sdk.NewInt(10))
+	suite.FundAcc(suite.TestAccs[0], defaultAcctFunds)
+
+	// Setup active pool
+	activePoolId := suite.PrepareBalancerPool()
+
+	// Setup mock inactive pool
+	k := suite.App.GAMMKeeper
+	ctrl := gomock.NewController(suite.T())
+	defer ctrl.Finish()
+	inactivePool := mocks.NewMockPoolI(ctrl)
+	inactivePoolId := k.GetNextPoolNumberAndIncrement(suite.Ctx)
+	inactivePool.EXPECT().IsActive(suite.Ctx).Return(false).AnyTimes()
+	inactivePool.EXPECT().GetId().Return(inactivePoolId).AnyTimes()
```

Comment:
Consider adding comments for these! imo could be really helpful since we're not very famialiar with how mock package works! (Also consider adding more detailed comments on how we mock a pool in the comment for the test itself!)